kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -0,0 +1,603 @@
+		thenNoExceptionsHaveBeenThrown();
+		thenMapResultShouldEqualInternalMap();
+	}
+
+	@Test
+	public void buildIpv4SettingsShouldWorkWhenGivenExpectedMapAndDhcpIsFalseAndEnabledForLan() {
+		givenValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus("wlan0", false, "netIPv4StatusEnabledLAN");
+		givenExpectedValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus("wlan0", false,
+				"netIPv4StatusEnabledLAN");
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuildIpv4SettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, "wlan0");
+		thenNoExceptionsHaveBeenThrown();
+		thenMapResultShouldEqualInternalMap();
+	}
+
+	@Test
+	public void buildIpv4SettingsShouldWorkWhenGivenExpectedMapAndDhcpIsFalseAndUnmanaged() {
+		givenValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus("wlan0", false, "netIPv4StatusUnmanaged");
+		givenExpectedValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus("wlan0", false,
+				"netIPv4StatusUnmanaged");
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuildIpv4SettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, "wlan0");
+		thenNoExceptionsHaveBeenThrown();
+		thenMapResultShouldEqualInternalMap();
+	}
+
+	@Test
+	public void build80211WirelessSettingsShouldWorkWhenGivenExpectedMapAndSetToInfraAndWithChannelField() {
+		givenValid80211WirelessSettingsWithTheFollowingParameters("wlan0", "testssid", "INFRA", true, false);
+		givenExpectedValid80211WirelessSettingsWithTheFollowingParameters("wlan0", "testssid", "infrastructure", true, false);
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuild80211WirelessSettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, "wlan0");
+		thenNoExceptionsHaveBeenThrown();
+		thenMapResultFromWifiSettingsShouldEqualInternalMapForWifiSettings();
+	}
+
+	@Test
+	public void build80211WirelessSecuritySettingsShouldWorkWhenGivenExpectedMap() {
+		givenValid80211WirelessSecuritySettingsWithTheFollowingParameters("wlan0", "ssidtest", "propmode", true, true);
+		givenExpected80211WirelessSecuritySettingsWithTheFollowingParameters("wlan0", "ssidtest", "propmode", true, true);
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuild80211WirelessSecuritySettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, "wlan0");
+		thenNoExceptionsHaveBeenThrown();
+		thenMapResultShouldEqualInternalMap();
+	}
+
+	@Test
+	public void buildSettingsShouldWorkWithExpectedInputsConfiguredForWiFiUnmanged() {
+		
+		String netInterface = "wlan0";
+		String kuraNetType = "netIPv4StatusUnmanaged";
+		
+		givenValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus(netInterface, false, kuraNetType);
+		givenExpectedValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus(netInterface, false,
+				kuraNetType);
```

Comment:
These two should be inside a `given` statement like:

```suggestion
		givenNetInterface("wlan0");
		givenIPStatus("netIPv4StatusUnmanaged");
		givenValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus(this.netInterface, false, this.kuraIPStatus);
		givenExpectedValidWifiConfigurationWithInterfaceNameAndDhcpBoolAndNetStatus(this.netInterface, false,
				this.kuraIPStatus);
```